FreeRTOS/FreeRTOS-Plus-TCP
# Fix Clang warnings

Description
-----------
Fixes `-Wall`, `-Wextra` and `-Weverything` warnings. Fixes the majority of `-Wpedantic` warnings as well, however a few couldn't be corrected AND pedantic isn't officially supported in this library.

Heavily inspired by @phelter and his changes found [here](https://github.com/phelter/FreeRTOS-Plus-TCP/tree/feature/fixing-clang-gnu-compiler-warnings)

Test Steps
-----------
* Ran build checks locally on my Ubunto 18.04 instance with GCC 7.5.0.

Checklist:
----------
- [ x ] I have tested my changes. No regression in existing tests.
- [ x ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Second point is non-applicable.

Related Issue
-----------
https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/issues/558


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
